### ISSUE TITLE
508-defect-2 [COGNITION, SCREEN READER]: Important alerts should be announced to screen reader users

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#17817]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** Josh

## User Story or Problem Statement

As a screen reader user, I want to be made aware of new instructions on the page related to calling my representative so I know what to expect in the future.

## Details

The "We’ll call your representative 2 times" information alert is not announced to screen reader users when it appears in the form. As screen reader users would probably be tabbing through the form to complete it, they may skip over this important information.

## Environment

* Operating System: Mac, Windows
* Browser: Safari, Firefox
* Screenreading device: VoiceOver, NVDA
* Server destination: staging

## Steps to Recreate

1. Enter `https://staging.va.gov/decision-reviews/higher-level-review/request-higher-level-review-form-20-0996/request-informal-conference` in browser
2. Follow test case 3's instructions and reach Step 3 of 4: Request an informal conference.
3. Start up VoiceOver in Safari.
4. Select "Yes, call my representative."
5. Select a time.
6. Confirm the alert is not politely announced.

## Solution

This could be resolved in several different ways:

### A. Provide the information in the alert on its own separate page
- Users will be able to read through important information with less cognitive overload if this information is split apart from an already complicated page. 
- This method will also work better for screen reader users.
- Uncertain what the dev lift will be for implementing this. 

### B. Add `role="alert"` and `aria-live="polite"` OR `role="status"`
- Following the VA design system accessibility recommendations, we could use the ARIA role="alert" to inform assistive technologies of the important message.
- This would politely announce to the screen reader user that additional information has been added to the page.
- Uncertain if this is feasible with the design system implementation.

### C. Add `aria-describedby` to the input field
- By adding a unique id to the information alert and `aria-describedby` to the input, the user can hear the additional information as part of its description.
- Poor experience to listen to as the label is already verbose; likely most screen readers will skip over without hearing it.

### D. Change the information alert into an accordion hint
- Adds the information alert into the tab order naturally.
- Reduces the amount of space taken on an already complicated page.
- Depending on how its written, users may skip/ignore it as it doesn't have the same visual prominence of an alert.

Curious to get your thoughts @Mottie and @kevinstachura on what would be the best way forward given constraints.

## WCAG or Vendor Guidance

* [WCAG 2.1, 4.1.3 Status Messages AA](https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html)
* [VA Design System Alertbox Accessibility Guidance](https://design.va.gov/components/alertboxes#accessibility)

## Screenshots or Trace Logs

<img width="1792" alt="Screen Shot 2020-12-22 at 10 50 09 AM" src="https://user-images.githubusercontent.com/14154792/102907242-c4c9f700-4443-11eb-9b23-fd1e30d971df.png">

**Comment by Mottie** [comment]
OBE based on the new design that does not have any alerts

![](https://user-images.githubusercontent.com/136959/103584376-c74a3900-4ea6-11eb-8cef-3451d324ab75.png)

**Comment by joshkimux** [comment]
Closing as this has been addressed with the new page split!
